scripts/resolver.py

```diff
@@ -0,0 +1,111 @@
+from sly import Lexer, Parser
+from decimal import Decimal
+
+
+class RuleLexer(Lexer):
+    tokens = {
+        # tokens
+        'RULE',
+        'VARIABLE',
+        'NUMBER',
+
+        # special symbols
+        'AND',
+        'OR',
+        'EQUAL',
+        'GREATER_THAN_EQUAL',
+        'LESS_THAN_EQUAL',
+        'TERNARY'
+    }
+    ignore = ' \t'
+    literals = {'+', '-', '*', '/', '>', '<', '!', ':', '(', ')'}
+
+    # tokens
+    RULE = r'rule_([a-zA-z0-9]+)([^() ])*'
+    VARIABLE = r'([a-zA-z]+)([^() ])*'
+    NUMBER = r'-?\d+(\.\d+)?'
+
+    # special symbols
+    AND = r'&&'
+    OR = r'\|\|'
+    EQUAL = r'=='
+    GREATER_THAN_EQUAL = r'>='
+    LESS_THAN_EQUAL = r'<='
+    TERNARY = r'\?'
```

Comment:
"Ternary" is a general term that means "three operands". The `?` operator is a ternary operator, and it is called the conditional operator. Should call it that.